src/Sylius/Bundle/AddressingBundle/EventListener/ZoneMemberIntegrityListener.php

```diff
@@ -18,14 +18,15 @@
 use Sylius\Component\Addressing\Model\CountryInterface;
 use Sylius\Component\Addressing\Model\ZoneInterface;
 use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\HttpFoundation\RequestStack;
 use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
 use Symfony\Component\HttpFoundation\Session\SessionInterface;
 use Webmozart\Assert\Assert;
 
 final class ZoneMemberIntegrityListener
 {
     public function __construct(
-        private SessionInterface $session,
+        private SessionInterface|RequestStack $sessionOrRequestStack,
```

Comment:
Why don't we just use `RequestStack` here? It already has `getSession()` method in Symfony 5, so we will not have a problem with removing it later 😄

Reply:
I'm not sure about what you mean. If you talk about the argument type SessionInterface|RequestStack, we cannot getting session from RequestStack on Symfony 4.4.
If you talk about the argument name "$requestStack", I agree, we can use this name to make it simpler while dropping Symfony 4 support (next release).